From 28356d8926bb4fdc0143ae53d8cbb02b49263390 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Sun, 8 Aug 2021 22:43:32 -0800
Subject: [PATCH 3313/4585] [Huawei] sched: Fix branch prediction error in
 static_key

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 175551

--------------------------------

sched_blocked_averages is trun on in most cases, it should be static_branch_likely
rather than static_branch_unlikely.

Signed-off-by: Yu Jiahua <yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=17763
Signed-off-by: Zheng Zucheng <zhengzucheng@huawei.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 022ac0addabc..7df7b63feb23 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -8971,7 +8971,7 @@ static void update_blocked_averages(int cpu)
 	update_rq_clock(rq);
 
 #ifdef CONFIG_SCHED_OPTIMIZE_LOAD_TRACKING
-	if (!static_branch_unlikely(&sched_blocked_averages)) {
+	if (!static_branch_likely(&sched_blocked_averages)) {
 		rq_unlock_irqrestore(rq, &rf);
 		return;
 	}
-- 
2.17.1

